File: LeetCode/Binary Tree Level Order Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    void bfs(TreeNode* root, vector<vector<int>> &ans) {
        if (!root) return;
        
        queue<TreeNode*> q;
        q.push(root);
        
        while(!q.empty()) {
            vector<int> siblings;
            TreeNode* node = q.front();
            q.pop();
            
            int size = q.size();
            siblings.push_back(node->val);
            
            if (node->left){
                q.push(node->left);
            }
        
            if (node->right){
                q.push(node-> right);
            }
            
            while(size--) {
                TreeNode* curr = q.front();
                q.pop();
                siblings.push_back(curr->val);
                
                if (curr->left){
                    q.push(curr->left);
                }
        
                if (curr->right){
                    q.push(curr->right);
                }   
            }
            
            if (!siblings.empty())
                ans.push_back(siblings);
        }
    }
    
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> ans;
        bfs(root, ans);
        
        return ans;
    }
};

File: LeetCode/Binary Tree Level Order Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def bfs(self, root):
        ans = []
        queue = [root]
        
        if (not root): return ans
        
        while(len(queue) > 0):
            level = []
            
            for i in range(len(queue)):
                current = queue.pop(0)
                level.append(current.val)
                
                if(current.left):
                    queue.append(current.left)
                
                if(current.right):
                    queue.append(current.right)
                    
            ans.append(level)
        
        return ans
        
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        return self.bfs(root)
        